Classes/Private/Wrapper/OCTToxAV.m

```diff
@@ -7,7 +7,298 @@
+
+    toxav_callback_call(_toxAV, callIncomingCallback, (__bridge void *)(self));
+    toxav_callback_call_state(_toxAV, callstateCallback, (__bridge void *)(self));
+    toxav_callback_audio_bit_rate_status(_toxAV, audioBitRateStatusCallback, (__bridge void *)(self));
+    toxav_callback_video_bit_rate_status(_toxAV, videoBitRateStatusCallback, (__bridge void *)(self));
+    toxav_callback_audio_receive_frame(_toxAV, receiveAudioFrameCallback, (__bridge void *)(self));
+    toxav_callback_video_receive_frame(_toxAV, receiveVideoFrameCallback, (__bridge void *)(self));
+
+    return self;
+}
+
+- (void)start
+{
+    DDLogVerbose(@"%@: start method called", self);
+
+    @synchronized(self) {
+        if (self.timer) {
+            DDLogWarn(@"%@: already started", self);
+            return;
+        }
+
+        dispatch_queue_t queue = dispatch_queue_create("me.dvor.objcTox.OCTToxAVQueue", NULL);
+        self.timer = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, queue);
+
+        uint64_t interval = toxav_iteration_interval(self.toxAV) * (NSEC_PER_SEC / 1000);
+        dispatch_source_set_timer(self.timer, dispatch_walltime(NULL, 0), interval, interval / 5);
+
+        __weak OCTToxAV *weakSelf = self;
+        dispatch_source_set_event_handler(self.timer, ^{
+            OCTToxAV *strongSelf = weakSelf;
+            if (! strongSelf) {
+                return;
+            }
+
+            toxav_iterate(strongSelf.toxAV);
+        });
+
+        dispatch_resume(self.timer);
+    }
+    DDLogInfo(@"%@: started", self);
+}
+
+- (void)stop
+{
+    DDLogVerbose(@"%@: stop method called", self);
+
+    @synchronized(self) {
+        if (! self.timer) {
+            DDLogWarn(@"%@: toxav isn't running, nothing to stop", self);
+            return;
+        }
+
+        dispatch_source_cancel(self.timer);
+        self.timer = nil;
+    }
+
+    DDLogInfo(@"%@: stopped", self);
+}
+
+- (void)dealloc
+{
+    [self stop];
+    toxav_kill(self.toxAV);
+    DDLogVerbose(@"%@: dealloc called, toxav killed", self);
+}
+
+#pragma mark - Call Methods
+
+- (BOOL)callFriendNumber:(OCTToxFriendNumber)friendNumber audioBitRate:(OCTToxAVAudioBitRate)audioBitRate videoBitRate:(OCTToxAVVideoBitRate)videoBitRate error:(NSError **)error
+{
+    NSParameterAssert(friendNumber);
+
+    TOXAV_ERR_CALL cError;
+    BOOL status = toxav_call(self.toxAV, friendNumber, audioBitRate, videoBitRate, &cError);
+
+    [self fillError:error withCErrorCall:cError];
+
+    return status;
+}
+
+#pragma mark - Private
+
+- (void)fillError:(NSError **)error withCErrorInit:(TOXAV_ERR_NEW)cError
+{
+    if (! error || (cError == TOXAV_ERR_NEW_OK)) {
+        return;
+    }
+
+    OCTToxAVErrorInitCode code = OCTToxAVErrorInitCodeUnknown;
+    NSString *description = @"Cannot initialize ToxAV";
+    NSString *failureReason = nil;
+
+    switch (cError) {
+        case TOXAV_ERR_NEW_OK:
+            NSAssert(NO, @"We shouldn't be here!");
+            return;
```

Comment:
Actually `break` is better than `return`. In case if asserts will be off we will still get an error.